src/com/gfredericks/test/chuck/clojure_test.cljc

```diff
@@ -106,27 +105,45 @@
   (ct/report reports))
 
 (defmacro checking
-  "A macro intended to replace the testing macro in clojure.test with a
+  {:doc      "A macro intended to replace the testing macro in clojure.test with a
   generative form. To make (testing \"doubling\" (is (= (* 2 2) (+ 2 2))))
   generative, you simply have to change it to
-  (checking \"doubling\" 100 [x gen/int] (is (= (* 2 x) (+ x x)))).
+  (checking \"doubling\" [x gen/int] (is (= (* 2 x) (+ x x)))).
 
-  You can optionally pass in a map options instead of the number of tests,
+  You can optionally pass the same options as test.check's defspec,
   which will be passed to `clojure.test.check/quick-check`, e.g.:
 
-    (checking \"doubling\" {:num-tests 100 :seed 123 :max-size 10}
+    (checking \"doubling\" 100 ;; number
+      [x gen/int]
+      (is (= (* 2 x) (+ x x))))
+
+    (checking \"doubling\" {:num-tests 100 :seed 123 :max-size 10} ;; options map
       [x gen/int]
       (is (= (* 2 x) (+ x x))))
 
   For background, see
   http://blog.colinwilliams.name/blog/2015/01/26/alternative-clojure-dot-test-integration-with-test-dot-check/"
-  [name num-tests-or-options bindings & body]
-  `(-testing ~name
-    (fn []
-      (let [final-reports# (atom [])]
-        (qc-and-report-exception final-reports# ~num-tests-or-options ~bindings ~@body)
-        (doseq [r# @final-reports#]
-          (-report r#))))))
+   :arglists '([name bindings & body] [name num-tests-or-options bindings & body])}
+  [name & check-decl]
+  (let [[num-tests-or-options bindings body]
+        (cond
+          (and (or (number? (first check-decl))
+                   (map? (first check-decl)))
+               (vector? (second check-decl)))
+          [(first check-decl) (second check-decl) (nnext check-decl)]
+
+          (vector? (first check-decl))
+          [nil (first check-decl) (next check-decl)]
+
+          :else (throw (#?(:clj  IllegalArgumentException.
+                           :cljs js/Error.) "Arguments to `checking` must be either [name bindings & body] or [name num-tests-or-options bindings & body]")))
+        num-tests-or-options (tc.clojure-test/process-options num-tests-or-options)]
```

Comment:
the fact that you're calling `process-options` with the form at macroexpansion-time means that users can't have a non-literal expression; e.g., I think `(checking "something" (assoc some-defaults :num-tests 40) ...)` would fail (you could add a test for this).

Now that I think about that distinction, I believe your check for `number?` and `map?` farther up are not sufficient, since it could also be a symbol or sequence. I think the cleanest thing would just be to only check `(vector? (second check-decl))` there.

Reply:
Done, but... (read the comment below)